### PR TITLE
[SourceKit] Migrate rename test to only check rename ranges and not the application of the rename

### PR DESCRIPTION
**Pull request by ahoppen**
In an upcoming PR, I will remove the rename refactoring actions that apply the rename and return the renamed source code. Instead, clients should call into the sourcekitd requests that return the ranges to rename.

To keep that upcoming PR smaller, I’m migrating all the test cases to use `find-(local-)rename-ranges` now.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

File: test/SourceKit/Refactoring/semantic-refactoring/local-rename.swift [hunk 1]
```diff
@@ -12,11 +12,24 @@ func foo() {
 }
 
 // RUN: %empty-directory(%t.result)
-// RUN: %sourcekitd-test -req=refactoring.rename.local -pos=2:8 -name new_name %s -- %s > %t.result/local-rename.swift.expected
-// RUN: %diff -u %S/local-rename.swift.expected %t.result/local-rename.swift.expected
-// RUN: %sourcekitd-test -req=find-local-rename-ranges -pos=2:8 %s -- %s > %t.result/local-rename-ranges.swift.expected
-// RUN: %diff -u %S/local-rename-ranges.swift.expected %t.result/local-rename-ranges.swift.expected
-// RUN: %sourcekitd-test -req=find-local-rename-ranges -pos=7:11 %s -- %s > %t.result/local-rename-lazy.swift.expected
-// RUN: %diff -u %S/local-rename-lazy.swift.expected %t.result/local-rename-lazy.swift.expected
+// RUN: %sourcekitd-test -req=find-local-rename-ranges -pos=2:8 -name new_name %s -- %s | %FileCheck %s --check-prefix CHECK-aa
```
Comment (bnbarham) [inline test/SourceKit/Refactoring/semantic-refactoring/local-rename.swift:15]: What's `name` used for?

File: test/refactoring/SyntacticRename/backticks-noncollapsible-separate-arglabel.swift [hunk 2]
```diff
@@ -5,4 +5,4 @@ Foo()/*test:ref*/[`x`: 10]
 
 // RUN: %empty-directory(%t.ranges)
 // RUN: %refactor -find-rename-ranges -source-filename %s -pos="test" -is-function-like -old-name "subscript(x:)" >> %t.ranges/backticks-noncollapsible-separate-arglabel.swift.expected
```
Comment (bnbarham) [inline test/refactoring/SyntacticRename/backticks-noncollapsible-separate-arglabel.swift:7]: Is `old-name` just used as a verification?

**Review by hamishknight — APPROVED** [review]

File: test/SourceKit/Refactoring/semantic-refactoring/local-rename.swift (hunk 1 above)
Comment (ahoppen) [reply test/SourceKit/Refactoring/semantic-refactoring/local-rename.swift:15]: It’s not used at all, AFAICT. I’ll remove it.

File: test/refactoring/SyntacticRename/backticks-noncollapsible-separate-arglabel.swift (hunk 2 above)
Comment (ahoppen) [reply test/refactoring/SyntacticRename/backticks-noncollapsible-separate-arglabel.swift:7]: Yes, used for verification that you don’t eg. rename an operator to a non-operator

https://github.com/apple/swift/blob/23effda44da844b450ae30b45ef721fe2508ee21/lib/Refactoring/SyntacticRename.cpp#L158-L195

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Review by bnbarham — APPROVED** [review]

**Pull request merged**